## Tarea 1: Análisis Exploratorio Satisfacción de Clientes

In [3]:
# Carga de datos
datos <- data.frame(
  Edad = c(25, 34, 28, 40, 35, 50, 23, 31, 27, NA, 54, 67, 56, 43, 33),
  Satisfaccion = c(5, 4, 3, 5, 4, 2, 5, 3, 4, NA, 4, 3, 4, 5, 1),
  Region = c("Norte", "Centro", "Sur", "Centro", "Norte", "Sur",
             "Centro", "Norte", "Centro", "Norte", "Norte", "Centro", 
             "Norte", "Sur", "Centro")
)

#### 1. Conclusiones de la Estructura de datos

In [4]:
str(datos)

'data.frame':	15 obs. of  3 variables:
 $ Edad        : num  25 34 28 40 35 50 23 31 27 NA ...
 $ Satisfaccion: num  5 4 3 5 4 2 5 3 4 NA ...
 $ Region      : chr  "Norte" "Centro" "Sur" "Centro" ...


**CONCLUSIONES DE LA ESTRUCTURA**

El dataset contiene 15 observaciones y 3 variables

- Edad: variable numérica (numeric) - CUANTITATIVA continua

- Satisfaccion: variable numérica (numeric) - ORDINAL (escala 1-5)

- Region: variable de texto (character) - CATEGÓRICA nominal

- Se detectan valores faltantes (NA) en Edad y Satisfaccion

- Región debería convertirse a factor para análisis categórico

#### 2. Conclusiones de Summary

In [5]:
summary(datos)

      Edad        Satisfaccion      Region         
 Min.   :23.00   Min.   :1.000   Length:15         
 1st Qu.:28.75   1st Qu.:3.000   Class :character  
 Median :34.50   Median :4.000   Mode  :character  
 Mean   :39.00   Mean   :3.714                     
 3rd Qu.:48.25   3rd Qu.:4.750                     
 Max.   :67.00   Max.   :5.000                     
 NA's   :1       NA's   :1                         

**CONCLUSIONES**

EDAD

- Rango: 23 a 67 años (amplitud de 44 años)

- Media: 38.21 años, Mediana: 34 años

- La media mayor que la mediana sugiere asimetría positiva (valores altos)

- 1 valor faltante

SATISFACCIÓN

- Escala completa utilizada: de 1 (mínimo) a 5 (máximo)

- Media: 3.714, Mediana: 4 - clientes moderadamente satisfechos

- La mediana de 4 indica que al menos 50% dio calificación ≥ 4

- 1 valor faltante (NA)


REGIÓN

- Norte: 6 clientes (mayor frecuencia)

- Centro: 6 clientes

- Sur: 3 clientes (menor frecuencia)

- Distribución desigual geográficamente

#### 3. Tabla de Frecuencias Absolutas y Relativas

In [11]:
# Tabla de frecuencias para Región
cat("=== REGIÓN ===\n")
tabla_region <- table(datos$Region)
freq_region <- data.frame(
    Frecuencia_Absoluta = as.vector(tabla_region),
    Frecuencia_Relativa = round(prop.table(tabla_region), 3),
    Porcentaje = paste0(round(prop.table(tabla_region) * 100, 1), "%")
)
rownames(freq_region) <- names(tabla_region)
print(freq_region)

cat("\n=== SATISFACCIÓN ===\n")
tabla_satisfaccion <- table(datos$Satisfaccion)
freq_satisfaccion <- data.frame(
    Frecuencia_Absoluta = as.vector(tabla_satisfaccion),
    Frecuencia_Relativa = round(prop.table(tabla_satisfaccion), 3),
    Porcentaje = paste0(round(prop.table(tabla_satisfaccion) * 100, 1), "%")
)
rownames(freq_satisfaccion) <- names(tabla_satisfaccion)
print(freq_satisfaccion)

cat("\n=== EDAD ===\n")
tabla_edad <- table(datos$Edad)
freq_edad <- data.frame(
    Frecuencia_Absoluta = as.vector(tabla_edad),
    Frecuencia_Relativa = round(prop.table(tabla_edad), 3),
    Porcentaje = paste0(round(prop.table(tabla_edad) * 100, 1), "%")
)
rownames(freq_edad) <- names(tabla_edad)
print(freq_edad)


=== REGIÓN ===


       Frecuencia_Absoluta Frecuencia_Relativa.Var1 Frecuencia_Relativa.Freq
Centro                   6                   Centro                      0.4
Norte                    6                    Norte                      0.4
Sur                      3                      Sur                      0.2
       Porcentaje
Centro        40%
Norte         40%
Sur           20%

=== SATISFACCIÓN ===
  Frecuencia_Absoluta Frecuencia_Relativa.Var1 Frecuencia_Relativa.Freq
1                   1                        1                    0.071
2                   1                        2                    0.071
3                   3                        3                    0.214
4                   5                        4                    0.357
5                   4                        5                    0.286
  Porcentaje
1       7.1%
2       7.1%
3      21.4%
4      35.7%
5      28.6%

=== EDAD ===
   Frecuencia_Absoluta Frecuencia_Relativa.Var1 Frecuencia_Relativa.Freq
2

**CONCLUSIONES**

**Región**

- La muestra está desbalanceada geográficamente
- Sur tiene la mitad de representación que las otras regiones
- Puede indicar: menor penetración de mercado, menor población, o sesgo en el muestreo
- Recomendación: Cualquier conclusión sobre diferencias regionales debe tomarse con cautela debido al tamaño desigual

**Satisfacción**

- Tendencia positiva: Más del 64% de clientes están satisfechos o muy satisfechos
- Pocos detractores: Solo 14% de calificaciones negativas (1-2)
- Distribución asimétrica negativa: Concentración hacia valores altos
- Área de mejora: 1 de cada 5 clientes es neutral (calificación 3) - potencial para mejorar

Señal de alerta:

Aunque hay pocos insatisfechos, existe 1 cliente con calificación 1 (muy insatisfecho)
Estos casos extremos requieren atención prioritaria para identificar problemas críticos

**Edad**

- Clientela heterogénea: Amplio espectro de edades
- No hay concentración etaria: Ningún grupo de edad domina claramente
- Diversidad demográfica: La empresa atiende desde jóvenes hasta adultos mayores
- Estrategia necesaria: Requiere enfoque de servicio multi-generacional

**SÍNTESIS GENERAL**

**Fortalezas:**

- Alto nivel de satisfacción general (64% satisfechos)
- Diversidad de clientes en edad
- Presencia en tres regiones

**Áreas de atención:**

- Desbalance regional: Sur está subrepresentada (¿oportunidad de expansión?)
- Clientes neutrales (21%): Potencial para convertir en satisfechos
- Casos críticos: 1 cliente muy insatisfecho necesita seguimiento
- Heterogeneidad etaria: ¿El servicio se adapta a todas las edades por igual?

**Recomendaciones:**

- Investigar por qué Sur tiene menos clientes
- Entrevistar al cliente con calificación 1 para identificar problemas críticos
- Analizar qué necesitan los clientes neutrales (calificación 3) para mejorar
- Segmentar análisis por edad y región para personalizar estrategias
- Ampliar muestra: 15 clientes es limitado para conclusiones definitivas

#### 4. Media y Mediana de la Variable Edad y Satisfacción

**CONSIDERACIONES**
- Ambas variables tienen valores NA que deben tratarse
- Se usará na.rm = TRUE para excluir NAs del cálculo
- Para Satisfacción (ordinal), la mediana es más apropiada que la media
- Los cálculos se basan en n=14 observaciones válidas\n

In [12]:
# Media y Mediana de Edad
cat("=== EDAD ===\n")
cat("Media:", mean(datos$Edad, na.rm = TRUE), "\n")
cat("Mediana:", median(datos$Edad, na.rm = TRUE), "\n")

# Media y Mediana de Satisfacción
cat("\n=== SATISFACCIÓN ===\n")
cat("Media:", mean(datos$Satisfaccion, na.rm = TRUE), "\n")
cat("Mediana:", median(datos$Satisfaccion, na.rm = TRUE), "\n")

=== EDAD ===
Media: 39 
Mediana: 34.5 

=== SATISFACCIÓN ===
Media: 3.714286 
Mediana: 4 


**Interpretación**

La **edad** promedio de los clientes es 38 años

- La media > mediana indica presencia de edades más altas que elevan el promedio (distribución asimétrica positiva)

La **satisfacción** típica es de 4 (mediana)

- En escalas ordinales, la MEDIANA es la medida más apropiada
- Satisfacción general: BUENA (4 de 5 puntos)

#### 5. Varianza y Desviación Estándar de la Edad

In [20]:
# Varianza y Desviación Estándar de Edad
cat("\n=== EDAD ===\n")
cat("Varianza:", var(datos$Edad, na.rm = TRUE), "\n")
cat("Desviación Estándar:", sd(datos$Edad, na.rm = TRUE), "\n")
# Coeficiente de Variación de Edad
cat("\n=== EDAD ===\n")
cat("Coeficiente de Variación:", 
    (sd(datos$Edad, na.rm = TRUE) / mean(datos$Edad, na.rm = TRUE)) * 100, 
    "%\n")



=== EDAD ===
Varianza: 176.4615 
Desviación Estándar: 13.28388 

=== EDAD ===
Coeficiente de Variación: 34.06124 %


**CONCLUSIONES**

- Desviación estándar: 13.28 años (ALTA dispersión)

- Coeficiente de variación: 34.06% (heterogeneidad marcada)

Interpretación: La empresa atiende a clientes muy diversos en edad, desde jóvenes hasta adultos mayores por lo que requiere estrategias de servicio diferenciadas por grupo etario